google/cloud/storage/storage_class.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_STORAGE_CLASS_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_STORAGE_CLASS_H_
+
+#include "google/cloud/storage/version.h"
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+/**
+ * Define functions with the well-know storage classes.
+ *
+ * This functions are provided to avoid typos when setting or comparing storage
+ * classes.
+ */
+namespace storage_class {
+inline char const* Standard() { return "STANDARD"; }
```

Comment:
The address of the return char* could change across invocations, leading to potential bugs (see the internal link at [1]).  To fix this, these should be written to return function-scoped `static constexpr` variables:

    inline char const* Standard() {
      static constexpr char kStorageClass[] = "STANDARD";
      return kStorageClass;
    }

[1] http://go/totw/140#string-view-mistake

Reply:
Comparing pointers has been a source of bugs since forever, and I am not sure I want to encourage it.  Changed these to return `std::string const&`, let me know what you think.

Reply:
True. And IIUC, the method I proposed in my last comment was supposed to allow a header-only definition (no .cc needed).